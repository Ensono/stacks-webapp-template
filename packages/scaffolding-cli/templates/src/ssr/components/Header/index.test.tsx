import {render} from "@testing-library/react"
import React from "react"
import {Header} from "."

const useRouter = jest.spyOn(require("next/router"), "useRouter")
jest.mock("../../lib/hooks", () => ({
    useUser: jest.fn().mockReturnValue({
        displayName: "",
        picture: "",
    }),
}))

const mockNextUseRouter = (props: {
    route: string
    pathname: string
    query: string
    asPath: string
}) => {
    useRouter.mockImplementationOnce(() => ({
        route: props.route,
        pathname: props.pathname,
        query: props.query,
        asPath: props.asPath,
    }))
}

test("renders Header snapshot", () => {
    mockNextUseRouter({
        route: "/",
        pathname: "/",
        query: "",
        asPath: "",
    })
    const {asFragment, getByTestId} = render(<Header />)
    expect(asFragment()).toMatchSnapshot()
    expect(getByTestId("create_button")).toBeInTheDocument()
})

test("renders no button on create page", () => {
    mockNextUseRouter({
        route: "/create",
        pathname: "/create",
        query: "",
        asPath: "",
    })
    const {asFragment, queryByTestId} = render(<Header />)
    expect(asFragment()).toMatchSnapshot()
    expect(queryByTestId("create_button")).toBeNull()
})
