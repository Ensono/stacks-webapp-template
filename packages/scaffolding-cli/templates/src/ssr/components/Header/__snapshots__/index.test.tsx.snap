// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="blogs menu button"
        class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-primary"
        tabindex="0"
        title="Blog"
        type="button"
      >
        <span
          class="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            data-testid="blogs_button"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9H9V9h10v2zm-4 4H9v-2h6v2zm4-8H9V5h10v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h2
        class="MuiTypography-root MuiTypography-h2"
        style="margin: 0px auto;"
      >
        <a
          class="css-15rvyqs"
        >
          Yumido
        </a>
      </h2>
      <button
        aria-label="create menu button"
        class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-primary"
        tabindex="0"
        title="Create menu"
        type="button"
      >
        <span
          class="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            data-testid="create_button"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`renders no button on create page 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="blogs menu button"
        class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-primary"
        tabindex="0"
        title="Blog"
        type="button"
      >
        <span
          class="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            data-testid="blogs_button"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9H9V9h10v2zm-4 4H9v-2h6v2zm4-8H9V5h10v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h2
        class="MuiTypography-root MuiTypography-h2"
        style="margin: 0px auto;"
      >
        <a
          class="css-15rvyqs"
        >
          Yumido
        </a>
      </h2>
    </div>
  </header>
</DocumentFragment>
`;
